### DISCUSSION TITLE
self-hosted setup throws an unexpected token 'export' exception

### DISCUSSION THREAD

**Discussion by Toharawk** [plausible/analytics#2987] [Self-Hosted Support]
I've added the code snippet to my application according to the documentation. Unfortunately, plausible returns an error the moment I try to read the analytics

<img width="2560" alt="Screenshot 2023-05-30 at 14 32 33" src="https://github.com/plausible/analytics/assets/52290471/ba0e1970-f133-4d53-a868-8c769d505ec3">
<img width="2560" alt="Screenshot 2023-05-30 at 14 32 54" src="https://github.com/plausible/analytics/assets/52290471/eedb3e6a-ee06-4344-b394-95ecae15a45d">
<img width="2560" alt="Screenshot 2023-05-30 at 14 42 17" src="https://github.com/plausible/analytics/assets/52290471/5fa6faf7-c0cc-47f9-bdb1-f50c45a14e73">

Did I miss something? Many thanks in advance!

**Comment by ruslandoga** [comment]
👋 @Toharawk 

Can you please share more info about your setup?

**Reply by Toharawk** [reply]
👋 @ruslandoga Certainly. I'm hosting plausible in a Debian environment. Please find the screenshot with setup details below:

<img width="1032" alt="Screenshot 2023-05-31 at 14 03 45" src="https://github.com/plausible/analytics/assets/52290471/b6dc85a9-6ebb-4a30-8f77-fd47c786aa14">

**Reply by ruslandoga** [reply]
Do you use docker compose config from https://github.com/plausible/hosting?

**Reply by Toharawk** [reply]
I do. As mentioned in #2980, I had to remove both lines from docker-compose to get it running. The rest of the configuration is unchanged.

**Reply by ruslandoga** [reply]
Seems like `fetch.min.js` comes from https://github.com/plausible/analytics/blob/6c907cbf4221788b31e78de77a26d32440feadf2/lib/plausible_web/templates/stats/waiting_first_pageview.html.eex#L1

**Reply by ruslandoga** [reply]
Related:

- https://github.com/github/fetch/issues/1027
- https://github.com/github/fetch/issues/791
- etc: https://github.com/github/fetch/issues?q=export

**Reply by ruslandoga** [reply]
I wonder what browser (and version) you are using. It seems like according to the issues and https://caniuse.com/?search=export it only fails on pretty old ones.

**Reply by ruslandoga** [reply]
https://github.com/plausible/analytics/pull/2999 removes `fetch.min.js` as it's probably no longer needed: https://caniuse.com/?search=fetch

**Reply by Toharawk** [reply]
Thanks! Does #2999 need to be applied to https://github.com/plausible/hosting first? The current build still throws the fetch.min.js exception.

**Reply by ruslandoga** [reply]
We'd need to release a new version and build a container image for it, yes.

In the meantime you can try using a different browser.

**Reply by ruslandoga** [reply]
#2999 has been merged and included in `v2.0.0-rc.2`: https://github.com/plausible/analytics/discussions/3011
